calendar-bundle/src/Picker/EventPickerProvider.php

```diff
@@ -19,11 +19,27 @@
 use Contao\CoreBundle\Picker\AbstractInsertTagPickerProvider;
 use Contao\CoreBundle\Picker\DcaPickerProviderInterface;
 use Contao\CoreBundle\Picker\PickerConfig;
+use Knp\Menu\FactoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Translation\TranslatorInterface;
 
 class EventPickerProvider extends AbstractInsertTagPickerProvider implements DcaPickerProviderInterface, FrameworkAwareInterface
 {
     use FrameworkAwareTrait;
 
+    /**
+     * @var Security
+     */
+    private $security;
+
+    public function __construct(Security $security, FactoryInterface $menuFactory, RouterInterface $router, TranslatorInterface $translator = null)
```

Comment:
I would prefer to add the new argument at the end, just like we did with `$uploadPath` in the file picker.

Reply:
🤷‍♂ I usually add *my* dependencies first, and the ones needed by the parent class thereafter.

Reply:
I guess both approaches are fine. Would you mind if I still changed it for consistency reasons?

Reply:
Not at all, feel free to change it 🙃

Reply:
No, @aschempp approach is not fine, it's **extremely** confusing if the class extending the parent has a different constructor argument order. We should not even consider using such a coding style. You'd have to juggle with arguments when debugging, you cannot use parent services (https://symfony.com/doc/current/service_container/parent_services.html#overriding-parent-dependencies), testing is harder and much more.

Reply:
I agree with you, however, since the translator is optional, we cannot add the new argument at the end. I have therefore left things unchanged.